source/_posts/2016-01-25-upgrading-to-ruby-on-rails-5-0.markdown

```diff
@@ -0,0 +1,226 @@
+```
+
+The error indicates that, for some reason, it can’t find the submit button with the name `Create Category`. A simple look at the page where this button is found shows us that right now the button has the name `Create category`, category in all lowercase.
+
+Looking for changes in the Rails code related to that, I find [this commit](https://github.com/rails/rails/commit/c74f9cc0e8b68697e0d1917b876ffdf404ade3fc). The new form seems fine to me but it’s breaking this type of test. 
+
+We change the tests in our application so that they look through `Create category` then do the same for the rest of the equivalent cases that fail and voila, the tests pass.
+
+With this last change, all of our application’s tests pass. After that, it’s a good idea to test the application by hand and there the following presents itself ...
+
+
+## Problem 5 - render partial of .html files escapes the content
+
+One case that our tests don’t cover: The header and footer show escaped on the screen (that’s to say, we see these templates’ source code directly in the browser!).
+
+Our header and footer are partials with the extension .html, not .html.erb. After debugging, I notice that the [Raw handler](https://github.com/rails/rails/blob/v5.0.0.beta1/actionview/lib/action_view/template/handlers/raw.rb) is being used to manage those partials and the Raw handler, as seen in the code, returns a String. The render partial is therefore returning a String instead of an OutputBuffer to the process of renderization of the application layout. However, the application layout assembles the HTML to be rendered as an OutputBuffer upon concatenating the String received from the render partial. Since a String is not HTML safe, all that partial’s content is left escaped, thus generating the problem in question.
+
+This began happening after [this commit](https://github.com/rails/rails/commit/4be859f0fdf7b3059a28d03c279f03f5938efc80). Previously, the ERB handler processed an .html file and returned an OutputBuffer. As the default handler is now the Raw handler, the Raw handler processes it and returns a String. This doesn’t mean that the change is completely incorrect, it simply means that changing the default handler presents this detail that must be fixed. 
+
+Not only were the .html files being processed by the ERB handler, but [also the .js files](https://github.com/rails/rails/issues/2394), among others. Naturally, it makes little sense to use the ERB handler in flat JavaScript files and it’s for these cases that [the Raw handler was incorporated](https://github.com/rails/rails/commit/8bea607265a2c9bb9bb2188b0a79089ca373b814).
+
+The issue is that the render of an .html file should always return an OutputBuffer, since by definition the content is HTML safe. Regardless of the context, be it render template, partial, or whatever, it’s because of that that [we introduced a new HTML handler](https://github.com/rails/rails/commit/8a998b0fa7523c2c8eb6d0cf56e40408bf6e9b2e).
```

Comment:
I'm noticing that paragraph is probably odd in the Spanish version, specially where the second sentence begins.

We could say:
`
The issue is that the render of an .html file should always return an OutputBuffer, since by definition the content is HTML safe, regardless of the context, be it render template, partial, or whatever. It’s because of that that we introduced a new HTML handler.
`

However, I think I would leave only the following (removing the `regardless of the context ...`):

`
The issue is that the render of an .html file should always return an OutputBuffer, since by definition the content is HTML safe. It’s because of that that we introduced a new HTML handler.
`